### ISSUE TITLE
`scan_parquet` fails with `Generic LocalFileSystem error: Out of range of file` using `POLARS_FORCE_ASYNC=1`

### ISSUE THREAD

**Issue by nameexhaustion** [pola-rs/polars#12297]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

path = ".env/x.parquet" 
pl.DataFrame({"x": range(100)}).write_parquet(path, row_group_size=5)

print(pl.scan_parquet(path).collect())

```

And run using `POLARS_FORCE_ASYNC=1`

### Log output

```shell
polars.exceptions.ComputeError: Generic LocalFileSystem error: Out of range of file /home/ubuntu/git/polars/.env/x.parquet, expected: 1560, actual: 1525
```

### Issue description

This issue began appearing on polars versions greater than `0.19.12-rc.1`.

### Expected behavior

Successfully reads the file.

### Installed versions

Tested on `main` @ `b5148bd6e`

<details>

```
--------Version info---------
Polars:              0.19.13-rc.1
Index type:          UInt32
Platform:            Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Python:              3.11.6 (main, Oct 25 2023, 21:49:07) [GCC 9.4.0]

----Optional dependencies----
adbc_driver_sqlite:  0.7.0
cloudpickle:         3.0.0
connectorx:          0.3.2
deltalake:           0.10.1
fsspec:              2023.10.0
gevent:              23.9.1
matplotlib:          3.8.0
numpy:               1.26.1
openpyxl:            3.1.2
pandas:              2.1.2
pyarrow:             13.0.0
pydantic:            2.4.2
pyiceberg:           0.5.0
pyxlsb:              1.0.10
sqlalchemy:          2.0.22
xlsx2csv:            0.8.1
xlsxwriter:          3.1.9
```

</details>

**Comment by ritchie46** [comment]
Is this fixed by #12308?

**Comment by ritchie46** [comment]
Alrigth, taking a look.
